test/framework/ownerreference_helpers.go

```diff
@@ -168,7 +169,7 @@ var CoreOwnerReferenceAssertion = map[string]func([]metav1.OwnerReference) error
 	},
 	machineKind: func(owners []metav1.OwnerReference) error {
 		// Machines must be owned and controlled by a MachineSet or a KubeadmControlPlane, depending on if this Machine is part of a ControlPlane or not.
-		return HasOneOfExactOwners(owners, []simpleOwnerRef{machineSetController}, []simpleOwnerRef{kubeadmControlPlaneController})
```

Comment:
This param which is unexported was the main problem, but I think the refactor to just using ordinary ownerRefs with a custom string method makes this simpler.

Reply:
Sounds reasonable